cleanlab_studio/studio/trustworthy_language_model.py

```diff
@@ -72,307 +78,383 @@ def __init__(
+        Get response and trustworthiness score from TLM.
+        The list returned will have the same length as the input list, if there are any
+        failures (errors or timeout) processing some inputs, the list will contain None
+        in place of the response.
 
-        else:
-            raise ValueError("prompt must be a string or list of strings.")
+        Args:
+            prompt (Sequence[str]): list of multiple prompts for the TLM
+        Returns:
+            List[Optional[TLMResponse]]: list of [TLMResponse](#class-tlmresponse) objects containing the response and trustworthiness score.
+                Entries of the list will be None for prompts that fail (due to any errors or timeout).
+        """
+        validate_tlm_try_prompt(prompt)
+
+        return self._batch_prompt_with_handled_errors(prompt)
 
     async def prompt_async(
+        self,
+        prompt: Union[str, Sequence[str]],
+        /,
+    ) -> Union[TLMResponse, List[TLMResponse]]:
+        """
+        (Asynchronously) Get response and trustworthiness score from TLM.
+
+        Args:
+            prompt (str | Sequence[str]): prompt (or list of multiple prompts) for the TLM
+        Returns:
+            TLMResponse | List[TLMResponse]: [TLMResponse](#class-tlmresponse) object containing the response and trustworthiness score.
+                    If multiple prompts were provided in a list, then a list of such objects is returned, one for each prompt.
+        """
+        validate_tlm_prompt(prompt)
+
+        async with aiohttp.ClientSession() as session:
+            if isinstance(prompt, str):
+                return await asyncio.wait_for(
+                    self._prompt_async(prompt, session), timeout=self._timeout
+                )
+
+            return cast(
+                List[TLMResponse],
+                await self._batch_async([self._prompt_async(p, session) for p in prompt]),
+            )
+
+    async def _prompt_async(
         self,
         prompt: str,
-        options: Optional[TLMOptions] = None,
         client_session: Optional[aiohttp.ClientSession] = None,
-        retries: int = 0,
     ) -> TLMResponse:
         """
-        (Asynchronously) Get response and confidence from TLM.
+        Private asynchronous method to get response and trustworthiness score from TLM.
 
         Args:
             prompt (str): prompt for the TLM
-            options (Optional[TLMOptions]): options to parameterize TLM with. Defaults to None.
-            client_session (Optional[aiohttp.ClientSession]): async HTTP session to use for TLM query. Defaults to None.
-            retries (int): number of retries for TLM query. Defaults to 0.
         Returns:
-            TLMResponse: [TLMResponse](#class-tlmresponse) object containing the response and confidence score
+            TLMResponse: [TLMResponse](#class-tlmresponse) object containing the response and trustworthiness score
         """
+
         async with self._query_semaphore:
```

Comment:
need to move semaphore / limiting to inside `tlm_prompt` (to properly handle retries)